packages/contentful--app-scripts/lib/get-management-token.js

```diff
@@ -3,37 +3,65 @@
 const chalk = require('chalk');
 const open = require('open');
 const inquirer = require('inquirer');
+const { cacheEnvVars } = require('../utils/cache-credential');
+const { createClient } = require('contentful-management');
+const {
+  ACCESS_TOKEN_ENV_KEY
+} = require('../utils/constants');
+
+const checkTokenValidity = async (accessToken) => {
+  try {
+    const client = createClient({ accessToken });
+    await client.getOrganizations();
+    return true;
+
+  } catch(err) {
+    return false;
+  }
+}
 
 async function getManagementToken() {
   const redirectUrl = 'https://www.contentful.com/developers/cli-oauth-page/';
   const CLIENT_ID = '9f86a1d54f3d6f85c159468f5919d6e5d27716b3ed68fd01bd534e3dea2df864';
   const oauthUrl = `https://be.contentful.com/oauth/authorize?response_type=token&scope=content_management_manage&client_id=${CLIENT_ID}&&redirect_uri=${encodeURIComponent(
     redirectUrl
   )}`;
-  try {
-    open(oauthUrl);
-  } catch (err) {
-    console.log(`${chalk.red('Error:')} Failed to open browser`);
-    console.log(err.message);
-    throw err;
-  }
 
-  const { mgmtToken } = await inquirer.prompt([
-    {
-      name: 'mgmtToken',
-      message: 'Please paste your access token:',
-      type: 'password',
-      validate(answer) {
-        if (!answer) {
-          return `${chalk.red('Error:')} Failed to login into Contentful.`;
-        }
-
-        return true;
+  const cachedAccessToken = process.env[ACCESS_TOKEN_ENV_KEY];
+  const cachedTokenValid = await checkTokenValidity(cachedAccessToken);
+
+  if (cachedTokenValid) {
+    return cachedAccessToken;
+
+  } else {
+
+    try {
+      open(oauthUrl);
+    } catch (err) {
+      console.log(`${chalk.red('Error:')} Failed to open browser`);
+      console.log(err.message);
+      throw err;
+    }
+
+    const { mgmtToken } = await inquirer.prompt([
+      {
+        name: 'mgmtToken',
+        message: 'Please paste your access token:',
+        type: 'password',
+        validate(answer) {
+          if (!answer) {
+            return `${chalk.red('Error:')} Failed to login into Contentful.`;
+          }
+
+          return true;
+        },
       },
-    },
-  ]);
+    ]);
+
+    await cacheEnvVars({[ACCESS_TOKEN_ENV_KEY]: mgmtToken});
```

Comment:
Just an idea, wouldn't it make sense to somehow validate the token before caching it? Otherwise it would lead to cache invalid tokens. But it might also be a bit too much

Reply:
We only cache the token pasted in. We can check if it's non-empty but I'm not sure if this is what you mean?

Reply:
Yes, but what would happen currently is that we ask for the token and cache it even if it's invalid, then the request fails. 
When the user retries the script the cached token is then picked up and fails at that validation so it will be asked again. 
I'm just wondering if we could save that step to not cache it if it's invalid beforehand. 
But in the end, it probably is not really critical, as the number of requests are the same, so I would be fine with both :)